### ISSUE TITLE
Create a new Test for project main deliverables - Zone, Auth, Char each.

### ISSUE THREAD

**Issue by sagunkho** [horizonxyz/horizon#169]
Create a new Test for project main deliverables. For example, test the functions of all Zone startup and shutdown -
```cpp
BOOST_AUTO_TEST_CASE(SCENARIO_LOGIN_TEST)
{
	sZone->general_conf().set_test_run();

	sZone->general_conf().set_config_file_path("../../../../../config/zone-server.lua.dist");
	sZone->read_config();
	
	sZone->config().set_static_db_path(std::string("../../../../../db/"));
	HLog(info) << "Static database path set to " << sZone->config().get_static_db_path() << "";

	sZone->config().set_mapcache_path(std::string("../../../../../db/maps.dat"));
	HLog(info) << "Mapcache file name is set to " << sZone->config().get_mapcache_path() << ", it will be read while initializing maps.";

	sZone->config().set_script_root_path(std::string("../../../../../scripts/"));

	sZone->initialize();

	std::cout << "Test case ending..." << std::endl;

	sZoneEnd;

	std::cout << "Test case ended." << std::endl;
}
```

create a test case for every scenario as well.

linked pull requests - #148

**Comment by sagunkho** [comment]
Partially Fixed in [2d06a73](https://github.com/horizonxyz/horizon/pull/148/commits/2d06a733f82016cbf7563eed162e4170e0e86459)

**Comment by sagunkho** [comment]
Added Auth and Char Server Tests
